apt/minimization/minimizer.py

```diff
@@ -340,7 +341,7 @@ def fit(self, X: Optional[DATA_PANDAS_NUMPY_TYPE] = None, y: Optional[DATA_PANDA
                 generalized = self._generalize_from_generalizations(x_test, self.generalizations)
 
             # check accuracy
-            accuracy = self.estimator.score(ArrayDataset(self.encoder.transform(generalized), y_test))
+            accuracy = self.estimator.score(ArrayDataset(self.encoder.transform(generalized).astype(dtype), y_test))
```

Comment:
Can the original data have a complex dtype, e.g. 
`arr_df = np.array([('a', 1.0), ('b', 2.0)], dtype=np.dtype('U1, f8'))`
?
Will this work in the case of OneHotEncoder?